microsoft/PowerToys
# [FancyZones] Remove resolution from "device-id"

## Summary of the Pull Request

**What is this about:**

Removed monitor resolution from the device id. 
Skip default layouts on `applied-layouts.json` saving. 

**Note**:  in some cases wrong layout could be applied.
Since we remove a part of the id, some ids in the `applied-layouts.json` could become identical. This could lead to the layout replacement with the other one, since we can't identify which layout was applied last. That's the reason why I decided to skip default layouts (anyway they'll be applied if needed). But if there are different custom or template layouts, it could be tricky to decide which one should be applied.

```
"applied-layouts": [
        {
            "device-id": "some-monitor-id_3000_2000_{00000000-0000-0000-0000-000000000000}",
            "applied-layout": {
                "uuid": "{C6D80181-D95A-476E-9A57-C8E2A9D3713A}",
                "type": "custom",
                "show-spacing": false,
                "spacing": 0,
                "zone-count": 3,
                "sensitivity-radius": 20
            }
        },
        {
            "device-id": "some-monitor-id_1920_1080_{00000000-0000-0000-0000-000000000000}",
            "applied-layout": {
                "uuid": "{A20AE166-F6AD-4202-8271-165BE2B26537}",
                "type": "custom",
                "show-spacing": false,
                "spacing": 0,
                "zone-count": 2,
                "sensitivity-radius": 20
            }
        }
    ]
```

->

```
"applied-layouts": [
        {
            "device": {
                "monitor": "some-monitor-id",
                "virtual-desktop": "{00000000-0000-0000-0000-000000000000}"
            },
            "applied-layout": {
                "uuid": "{C6D80181-D95A-476E-9A57-C8E2A9D3713A}",
                "type": "custom",
                "show-spacing": false,
                "spacing": 0,
                "zone-count": 3,
                "sensitivity-radius": 20
            }
        }
    ]
```

**What is included in the PR:** 

**How does someone test / validate:** 

* Start FancyZones verify layout is correct (reset case described above)
* Change layout (with shortcut or via Editor)
* Verify the right layout is applied
* Restart FancyZones, verify the layout is correct

## Quality Checklist

- [x] **Linked issue:** #14261

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
